src/main/osd/osd_elements.c

```diff
@@ -962,7 +962,7 @@ static void osdElementLinkQuality(osdElementParms_t *element)
 #ifdef USE_BLACKBOX
 static void osdElementLogStatus(osdElementParms_t *element)
 {
-    if (IS_RC_MODE_ACTIVE(BOXBLACKBOX)) {
+    if (IS_RC_MODE_ACTIVE(BOXBLACKBOX) || !isModeActivationConditionPresent(BOXBLACKBOX)) {
```

Comment:
Not sure about this. Why would I want an indication of blackbox not working when I don't have the mode set up to activate the logging anyway? If the intended use case is for the extremely rare use case of `blackbox_mode = always` then include that in the condition as well so the element only appears in this case. In other words:

```
    if (IS_RC_MODE_ACTIVE(BOXBLACKBOX) || (!isModeActivationConditionPresent(BOXBLACKBOX) && blackboxConfig()->mode == BLACKBOX_MODE_ALWAYS_ON)) {
```

Reply:
yes, good idea. i'll investigate further and update the PR.

Reply:
@etracer65 if I'm remembering and understanding correctly, the `BLACKBOX_MODE_ALWAYS_ON` is used to log continually, and not 'when armed' (`BLACKBOX_MODE_NORMAL`).   So using `BLACKBOX_MODE_ALWAYS_ON` would not be correct.

The idea is to know if the blackbox system is OK *before* you arm, when using `BLACKBOX_MODE_NORMAL` and when NOT using a switch for blackbox.

Background: On SD-card equipped drones people insert an SD card when they need to log, and their config is such that if one is present it will log on armed, for every flight.

I think the PR is correct as-is.

Reply:
This still doesn't make any sense. `!isModeActivationConditionPresent(BOXBLACKBOX)` check is true only if the user hasn't configured the BLACKBOX aux mode. So why would we want to present a warning about the SD card when the user can't enable logging anyway. If your goal is to present some kind of advance warning that the logging **may not work** if they try to activate it, then something like this would be more appropriate:

```
if (IS_RC_MODE_ACTIVE(BOXBLACKBOX) || (isModeActivationConditionPresent(BOXBLACKBOX) && !ARMING_FLAG(ARMED))) {
```

Then we show the element when they have logging enabled (BOXBLACKBOX on) or if disarmed and they *could* enable blackbox. However even with the final `else` clause that reports the log number needs to be suppressed in this disarmed state as it would report the previous log number and not really useful.

Reply:
i'll digest your comments and dig into it a bit more.  thanks for the review and feedback so far.